src/sinks/util/buffer/mod.rs

```diff
@@ -38,6 +38,15 @@ impl Compression {
             Self::Gzip => "log.gz",
         }
     }
+
+    #[cfg(feature = "rusoto_core")]
+    pub fn to_rusoto(&self) -> rusoto_core::encoding::ContentEncoding {
+        match self {
+            Self::None => rusoto_core::encoding::ContentEncoding::Identity,
+            // 6 is default, add Gzip level support to vector in future
+            Self::Gzip => rusoto_core::encoding::ContentEncoding::Gzip(None, 6),
+        }
+    }
```

Comment:
Would this be better handled as either an `impl From` or `impl Into`? I think `From` might run into the orphan rules, but one should work.

Reply:
This will be awesome. I added `From<Compression>`, but... maybe it's better implement `From<&Compression>`? From other side change to `From<&Compression>` looks like premature optimization. `Compression` is super simple and implement `Copy`...

Reply:
I see no point to implementing `From<&Compression>` when the type implements `Copy`.